netbox/netbox/settings.py

```diff
@@ -77,6 +79,9 @@
 EMAIL = getattr(configuration, 'EMAIL', {})
 ENFORCE_GLOBAL_UNIQUE = getattr(configuration, 'ENFORCE_GLOBAL_UNIQUE', False)
 EXEMPT_VIEW_PERMISSIONS = getattr(configuration, 'EXEMPT_VIEW_PERMISSIONS', [])
+GITHUB_REPOSITORY_API = getattr(configuration, 'GITHUB_REPOSITORY_API',
```

Comment:
This behavior should be disabled by default to avoid any surprises for existing installations. Let's set `GITHUB_REPOSITORY_API` to `None` by default, which will require users to take a direct action to enable the functionality.